include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_for.h

```diff
@@ -37,6 +37,28 @@ namespace dpl
 namespace __par_backend_hetero
 {
 
+template <bool __can_vectorize_brick, bool __can_brick_process_multiple_iters, typename... _Ranges>
```

Comment:
Does it make sense to instead accept a `typename __brick` and a `bool __attempt_vectorization`?
instead of 2 booleans?

It seems like the vectorization is either dictated directly by the brick, or turned off wholesale by the implementation path.  Then here within `__pfor_params` you can assume `__can_vectorize` and `__can_process_multiple_iters` exist and use them.  If we really want to get fancy we could even make a defaulted `brick_traits<brick>` struct which could provide defaults (`false`) for bricks which don't define these themselves.

I'm not sure this option is better, but it may better follow the logic of what is going on in the caller.

Reply:
I think passing the brick and a single boolean the most sense. I have renamed it `__enable_tuning` because it applies to both vectorization and processing multiple iterations per work-item. In the case `__enable_tuning` is false, we process one input per work-item to evenly distribute work for small problem sizes.

I also considered having a brick trait, but I favor the current approach where the these boolean variables are defined near the scalar and vectorized implementations in the brick itself.